### ISSUE TITLE
bug(API): Loadbalancers are not working as expected

### ISSUE THREAD

**Issue by dipankardas011** [kubesimplify/ksctl#107]
# Describe 🐞 

https://github.com/kubesimplify/ksctl/blob/a96beeb111da6db096e109cbf90bae4caf5a7e48/api/civo/main.go#L100

## Provider in effect
- [x] Azure
- [x] Civo
- [ ] Local
- [ ] AWS

# Reproduce 💻 ➡️  💻 
try to deploy the nginx or other loadbalancers

# Possible Fix & Expected behavior🔧 
currently unknown

# Screenshots :framed_picture:

**Desktop (please complete the following information):**
 - OS
   - [x] MacOS
   - [x] Windows
   - [x] Linux
 - ksctl version `<Version>`

# Additional context

**Comment by dipankardas011** [comment]
## possible solutions
Switch from HA proxy to https://kube-vip.io/ or https://www.k8gb.io/

**Comment by dipankardas011** [comment]
https://www.youtube.com/watch?v=U46hlF0Z3xs

**Comment by ashutosh887** [comment]
Let me work on this @dipankardas011 
Please assign

**Comment by dipankardas011** [comment]
> Let me work on this @dipankardas011 
> Please assign

Sure you can do research and have a plan how and what changes we need to have
By that time new architecture design should be ready for all of us to work

**Comment by ashutosh887** [comment]
Hey is the design we were planning ready? @dipankardas011

**Comment by dipankardas011** [comment]
@ashutosh887 yes the design is ready just  some minor fix will be their in design and also codebase you can also check the current progress in pr #123

**Comment by dipankardas011** [comment]
> Hey is the design we were planning ready? @dipankardas011

everything is ready you can work on this issue

**Comment by ashutosh887** [comment]
Thanks @dipankardas011 
Where can I find it?

**Comment by dipankardas011** [comment]
Currently check in k3s distribution

**Comment by dipankardas011** [comment]
https://github.com/kubernetes/kubeadm/blob/main/docs/ha-considerations.md
